sonarr/qualitydefinition.go

```diff
@@ -0,0 +1,75 @@
+package sonarr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"golift.io/starr"
+)
+
+// QualityDefinition is the /api/v3/qualitydefinition endpoint.
+type QualityDefinition struct {
+	ID        int64            `json:"id,omitempty"`
+	Weight    int64            `json:"weight"`
+	MinSize   float64          `json:"minSize"`
+	MaxSize   float64          `json:"maxSize"`
+	Title     string           `json:"title"`
+	Qualities []*starr.Quality `json:"items"`
+}
+
+// GetQualityDefinitions returns all configured quality definitions.
+func (s *Sonarr) GetQualityDefinitions() ([]*QualityDefinition, error) {
+	return s.GetQualityDefinitionsContext(context.Background())
+}
+
+func (s *Sonarr) GetQualityDefinitionsContext(ctx context.Context) ([]*QualityDefinition, error) {
+	var output []*QualityDefinition
+
+	if _, err := s.GetInto(ctx, "v3/qualityDefinition", nil, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(qualityDefinition): %w", err)
+	}
+
+	return output, nil
+}
+
+// GetQualityDefinition returns a single quality definition.
+func (s *Sonarr) GetQualityDefinition(qualityDefinitionID int) (*QualityDefinition, error) {
+	return s.GetQualityDefinitionContext(context.Background(), qualityDefinitionID)
+}
+
+func (s *Sonarr) GetQualityDefinitionContext(ctx context.Context, qualityDefinitionID int) (*QualityDefinition, error) {
+	var output *QualityDefinition
+
+	if _, err := s.GetInto(ctx, "v3/qualityDefinition/"+strconv.Itoa(qualityDefinitionID), nil, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(qualityDefinition): %w", err)
+	}
+
+	return output, nil
+}
+
+// UpdateQualityDefinition updates the quality definition.
+func (s *Sonarr) UpdateQualityDefinition(definition *QualityDefinition) (*QualityDefinition, error) {
+	return s.UpdateQualityDefinitionContext(context.Background(), definition)
+}
+
+func (s *Sonarr) UpdateQualityDefinitionContext(ctx context.Context,
+	definition *QualityDefinition) (*QualityDefinition, error) {
+	var (
+		body   bytes.Buffer
+		output QualityDefinition
+	)
+
+	if err := json.NewEncoder(&body).Encode(definition); err != nil {
+		return nil, fmt.Errorf("json.Marshal(qualityDefinition): %w", err)
+	}
+
+	if _, err := s.PutInto(ctx, "v3/qualityDefinition/"+strconv.Itoa(int(definition.ID)),
+		nil, &body, &output); err != nil {
```

Comment:
I'm not a huge fan of breaking an `if` into multiple lines for the sake of making short lines. You can do the same thing, with a clearer piece of code, by unscoping `err` (which I generally do), or by doing this:
```suggestion
	id := strconv.Itoa(int(definition.ID))
	if _, err := s.PutInto(ctx, "v3/qualityDefinition/"+id, nil, &body, &output); err != nil {
```

fwiw, in these small methods I don't mind not putting `err :=` into the `if` line. Sometimes these get long, and it's just cleaner and easier to read to rip that out and put it on its own line.

Some of the other `if` statements in this contribution are a bit long and could use similar treatment, even if they don't trip the linter. 

This all looks very good, thank you!